### PR TITLE
[Event Hubs Client] AMQP Cancellation Support

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to take advantage of the support for cancellation tokens that was added to the AMQP transport library, preserving existing timeout behavior by specifying an operation timeout for the link.

# References and Related

- [Event Hubs: Cancellation Token Support for AMQP Operations (#24082)](https://github.com/Azure/azure-sdk-for-net/issues/24082)

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/Azure.Messaging.EventHubs.sln [hunk 1]
```diff
@@ -84,9 +84,6 @@ Global
 		{FB10A6B3-88DD-4296-9241-53ECDF942842}.Release|x64.ActiveCfg = Release|Any CPU
 		{FB10A6B3-88DD-4296-9241-53ECDF942842}.Release|x64.Build.0 = Release|Any CPU
 		{FB10A6B3-88DD-4296-9241-53ECDF942842}.Release|x86.ActiveCfg = Release|Any CPU
-<<<<<<< HEAD
```
Comment (JoshLove-msft) [inline sdk/eventhub/Azure.Messaging.EventHubs/Azure.Messaging.EventHubs.sln:87]: Looks like a bad merge.

Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/Azure.Messaging.EventHubs.sln:87]: It sure does.   Something got cross-threaded yesterday, for sure.

**Review by JoshLove-msft — APPROVED** [review]
Looks good aside from merge issue.

**Comment by jsquire** [comment]
There seems to be an issue with linking the CI run to this pull request.   The pipeline run has passed and can be found [[here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1229387&view=results)].

**Pull request merged**